### PR TITLE
Enhanced RDoc for Range

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treated:
- #size
- #to_a
- #each
- #begin
- #end
- #first
- #last

### CODE REVIEW

File: range.c [hunk 1]
```diff
@@ -778,14 +778,16 @@ sym_each_i(VALUE v, VALUE arg)
 
 /*
  *  call-seq:
- *     rng.size                   -> num
+ *    size -> non_negative_integer
```
Comment (jeremyevans) [inline range.c:781]: `Infinity` (used in the example below) is not a non-negative integer, nor is `nil`.  Maybe `integer | nil | Infinity`? Other approaches may work.

File: range.c [hunk 2]
```diff
@@ -809,13 +811,15 @@ range_size(VALUE range)
 
 /*
  *  call-seq:
- *     rng.to_a                   -> array
- *     rng.entries                -> array
+ *    to_a -> array
  *
- *  Returns an array containing the items in the range.
+ *  Returns an array containing the elements in +self+, if a finite colletion;
```
Comment (jeremyevans) [inline range.c:816]: `colletion` -> `collection`

File: range.c [hunk 3]
```diff
@@ -809,13 +811,15 @@ range_size(VALUE range)
...
+ *  raises an exception otherwise.
+ *
+ *    (1..4).to_a     # => [1, 2, 3, 4]
+ *    (1...4).to_a    # => [1, 2, 3]
+ *    ('a'..'d').to_a # => ["a", "b", "c", "d"]
  *
- *    (1..7).to_a  #=> [1, 2, 3, 4, 5, 6, 7]
- *    (1..).to_a   #=> RangeError: cannot convert endless range to an array
```
Comment (jeremyevans) [inline range.c:818]: Probably should mention `to_a` is aliased as `entries`.

File: range.c [hunk 4]
```diff
@@ -982,11 +982,15 @@ range_each(VALUE range)
 
 /*
  *  call-seq:
- *     rng.begin    -> obj
+ *    begin -> object
```
Comment (jeremyevans) [inline range.c:985]: This is a case where we may want to use `self.begin`, as `begin` is a keyword and you cannot use it as a bare method call.

File: range.c [hunk 5]
```diff
@@ -982,11 +982,15 @@ range_each(VALUE range)
...
  *
- *  Returns the object that defines the beginning of the range.
+ *  Returns the object that defined the beginning of +self+,
```
Comment (jeremyevans) [inline range.c:987]: `defined` -> `defines`, since ranges are immutable.

File: range.c [hunk 6]
```diff
@@ -982,11 +982,15 @@ range_each(VALUE range)
...
+ *  if one was given; otherwise returns +nil+:
```
Comment (jeremyevans) [inline range.c:988]: I would drop this line.  `nil` is the implicit beginning of unbounded ranges: `(..1) == (nil..1) # => true`.

File: range.c [hunk 7]
```diff
@@ -998,12 +1002,16 @@ range_begin(VALUE range)
 
 /*
  *  call-seq:
- *     rng.end    -> obj
+ *    end -> object
```
Comment (jeremyevans) [inline range.c:1005]: Likewise, `self.end` may be better than `end`

File: range.c [hunk 8]
```diff
@@ -998,12 +1002,16 @@ range_begin(VALUE range)
...
+ *
+ *  Returns the object that defined the end of +self+,
```
Comment (jeremyevans) [inline range.c:1007]: `defined` -> `defines`

File: range.c [hunk 9]
```diff
@@ -998,12 +1002,16 @@ range_begin(VALUE range)
...
+ *  if one was given; otherwise returns +nil+:
```
Comment (jeremyevans) [inline range.c:1008]: Again, I would remove this line.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Thanks for this.  Looks mostly good, I just have a few change requests.

File: range.c (hunk 3 above)
Comment (BurdetteLamar) [reply range.c:818]: In my html, alias came up automagically.

> Also aliased as: entries

File: range.c (hunk 1 above)
Comment (BurdetteLamar) [reply range.c]: Fixed ('or', not '|', as elsewhere.

File: range.c (hunk 2 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 3 above)
Comment (BurdetteLamar) [reply range.c:818]: But I added it anyway.

File: range.c (hunk 4 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 5 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 6 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 7 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 8 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

File: range.c (hunk 9 above)
Comment (BurdetteLamar) [reply range.c]: Fixed.

**Review by jeremyevans — APPROVED** [review]
Looks good, thanks!

**Pull request merged**